### PR TITLE
Unpublish flow

### PR DESCRIPTION
**Pull request by jonathan-s**
With this PR you can create a workflow that is designed to unpublish assets / content. The workflow has a new boolean property `is_unpublishing` that can be selected when you create a workflow. 

Once content has been added to be moderated with an unpublishing workflow all content will be unpublished once that has been approved and a signal with all unpublished content will be sent that can be further processed as needed. 

- ~~Depends on #159~~ 
- Related to https://github.com/FidelityInternational/djangocms-pageadmin/pull/36. To be able to add a collection from pagecontent admin, that PR is needed. 
- ~~Currently blocked by #160~~ 
- VersionAdmin also contains a link to unpublish. Which should be removed See #165 
- ~~Related to #166, an action that should be used for unpublishing.~~
- Related to https://github.com/divio/djangocms-versioning-filer/pull/49, an action added to filer admin for unpublishing.

### CODE REVIEW

File: tests/test_admin_actions.py [hunk 1]
```diff
@@ -647,6 +725,7 @@ def test_signal_when_published(self):
             self.assertEquals(published_mr[0], self.moderation_request1)
             self.assertEquals(kwargs['workflow'], self.collection.workflow)
 
+    # TODO Andrew, how is it intended that collection status should change?
```
Comment (jonathan-s) [inline tests/test_admin_actions.py]: Ie, right now once all moderation requests are approved the collection status is changed to archived. When reading the tests, it looks like this should NOT be the case. 

Either way, I would suggest that changing how `collection.status` works is out of scope for this PR and it should be changed in new PR.

Comment (Aiky30) [reply tests/test_admin_actions.py]: I think this is part of the Versioning state machine where versions state change are forced to go through state changes.

Comment (jonathan-s) [reply tests/test_admin_actions.py]: The versioning state machine and the collection state machine (albeit simpler than versioning) are pretty much separate. 

The behaviour of the collection state machine is basically the following. 
```
COLLECTING = "COLLECTING"
IN_REVIEW = "IN_REVIEW"
ARCHIVED = "ARCHIVED"
CANCELLED = "CANCELLED"
```

Where there seems to a be bug is that as soon as all requests has been approved the collection state changes from `in review` to `archived`. An archived state suggests that no more actions needs to be taken. Even though the collection is now archived there are still actions to be taken. 

I'll see if I manage to make a separate PR around that. It's out of scope of this PR anyway.

**Comment by jonathan-s** [comment]
I hate isort with a passion.

**Pull request closed without merging**